src/fides/api/alembic/migrations/versions/c9c72b3d550b_data_migration_converting_existing_.py

```diff
@@ -0,0 +1,214 @@
+            ),
+            id=str(uuid.uuid4()),
+            created_at=asset.created_at,
+            updated_at=asset.updated_at,
+            name=asset.name,
+            domain=asset.domain,
+            system_id=asset.system_id,
+            data_uses=asset.data_uses,
+            asset_type="Cookie",
+        )
+
+    # Delete the cookies table
+    logger.debug("Deleting cookies table")
+    connection.execute(sa.text("DROP TABLE cookies"))
+
+
+def downgrade():
+    op.drop_column("asset", "description")
+
+    # Recreate the cookies table
+    logger.debug("Recreating cookies table")
+    op.create_table(
+        "cookies",
+        sa.Column("id", sa.String(length=255), nullable=False),
+        sa.Column(
+            "created_at",
+            sa.DateTime(timezone=True),
+            server_default=sa.text("now()"),
+            nullable=True,
+        ),
+        sa.Column(
+            "updated_at",
+            sa.DateTime(timezone=True),
+            server_default=sa.text("now()"),
+            nullable=True,
+        ),
+        sa.Column("name", sa.String(), nullable=False),
+        sa.Column("domain", sa.String(), nullable=True),
+        sa.Column("path", sa.String(), nullable=True),
+        sa.Column("system_id", sa.String(), nullable=True),
+        sa.Column("privacy_declaration_id", sa.String(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["privacy_declaration_id"], ["privacydeclaration.id"], ondelete="SET NULL"
+        ),
+        sa.ForeignKeyConstraint(["system_id"], ["ctl_systems.id"], ondelete="CASCADE"),
+        sa.PrimaryKeyConstraint("id"),
+        sa.UniqueConstraint(
+            "name", "privacy_declaration_id", name="_cookie_name_privacy_declaration_uc"
+        ),
+    )
+    op.create_index(op.f("ix_cookies_id"), "cookies", ["id"], unique=False)
+    op.create_index(op.f("ix_cookies_name"), "cookies", ["name"], unique=False)
+    op.create_index(
+        op.f("ix_cookies_privacy_declaration_id"),
+        "cookies",
+        ["privacy_declaration_id"],
+        unique=False,
+    )
+    op.create_index(
+        op.f("ix_cookies_system_id"), "cookies", ["system_id"], unique=False
+    )
+    # ### end Alembic commands ###
+
+    connection = op.get_bind()
+    result = connection.execute(
+        sa.text(
+            "SELECT id, name, domain, system_id, data_uses FROM asset WHERE asset_type = 'Cookie'"
+        )
+    )
+
+    logger.debug("Migrating existing assets to cookies")
+    for row in result:
+        # Find the privacy declaration ID matching the system ID and data use
+        privacy_declaration_ids = []
+        for data_use in row.data_uses:
+            pud = connection.execute(
+                sa.text(
+                    "SELECT id FROM privacydeclaration WHERE system_id = :system_id AND data_use = :data_use"
+                ),
+                system_id=row.system_id,
+                data_use=data_use,
+            ).fetchone()
```

Comment:
ok so here it may not be quite as simple to 'join' with the privacy declaration table. but that doesn't mean we should issue a query for every asset! that can be problematic. 

we can just query the full privacy declaration table up front, store that in memory, and do the "joining" logic (which is admittedly more complex here, since we don't have a direct FK link) using python. the key thing i'd want to avoid though is issuing a query for every asset that we're migrating - there could be thousands.

Reply:
Good point, done here: https://github.com/ethyca/fides/pull/5776/commits/54e4eac935d25653649ec9f2424b76047206c7bb#diff-9a71bb9fcb8781cd9b038e712d899749d3e99ec8344ce94cfa617e6dbb0785a6R165